### PR TITLE
Rework map window to take map size into account

### LINKED ISSUE #7248: Minimap is not centred
**Version: Original RCT

Two issues with the minimap:
- The mini map is not properly centred. This is especially obvious when making the map-window large.
- Clicking on the map does not center the view on exactly where you click

I have already fixed this. But I will have to do some more clean-up before I can send a PR.

**Screenshots / Video:**
screen shot 2018-03-08 at 21 36 58

### LINKED ISSUE #13294: Map window rendering cuts off pixels in corner of maps facing West & North
**OS:** Windows 10
**Version:** 0.3.1
**Commit/Build:** d287653

I just noticed that when opening the map window, the map isn't centered properly and is positioned a little too far to the left or top when it's facing West or North respectively, which cuts off several pixels on the corner of the map. This seems to happen in vanilla RCT2 as well.

- [Yes] Reproducible in RCT2 (vanilla)?
- [No] Specific to multiplayer?

**Steps to reproduce:**
1. Load a scenario and view the map window.
2. The map will be positioned either too far left or too far up top depending on the rotation, which cuts off a bit of the corner view.

**Screenshots / Video:**

Forest Frontiers:
image
Electric Fields:
image

### PR DESCRIPTION
**Pull request by AaronVanGeffen**
The map window on develop does not take the current map size into account, instead working with the internal 'technical' maximum size of 1001x1001 tiles. This is relatively slow, and unnecessary. This PR reworks this, such that the current map size is used instead.

In addition, the initial and minimum size is reworked to fit any reasonably sized map, while constraining it to screen sizes. The maximum dimensions of the window are also changed to fit the current map size. The minimum sizes haven't changed.

Maps that are smaller than the window size are now offset to the centre of the scrollview. This closes #7248.

Finally, the map corners are no longer cut off. This is done by removing the `-8` offset, which was present in vanilla. This fixes #13294, a vanilla bug.

Some changelog entries are probably in order, but let's reserve those for after testing.

This PR:
Forest Frontiers 2024-05-07 14-40-26

On develop:
Forest Frontiers 2024-05-07 14-41-07

### CODE REVIEW

File: src/openrct2-ui/windows/Map.cpp [hunk 1]
```diff
@@ -1346,11 +1361,13 @@ static constexpr ScreenCoordsXY MiniMapOffsets[] = {
             if (mainViewport == nullptr)
                 return;
 
-            auto offset = MiniMapOffsets[GetCurrentRotation()];
-            auto leftTop = ScreenCoordsXY{ (mainViewport->viewPos.x >> 5) + offset.x,
-                                           (mainViewport->viewPos.y >> 4) + offset.y };
-            auto rightBottom = ScreenCoordsXY{ ((mainViewport->viewPos.x + mainViewport->view_width) >> 5) + offset.x,
-                                               ((mainViewport->viewPos.y + mainViewport->view_height) >> 4) + offset.y };
+            auto offset = MiniMapOffsetFactors[GetCurrentRotation()];
+            auto leftTop = ScreenCoordsXY{ (mainViewport->viewPos.x >> 5) + offset.x * getTechnicalMapSize() - 8,
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/Map.cpp:1365]: I see the `- 8` pop up in quite a few places. Maybe something for a constant?

Comment (AaronVanGeffen) [reply src/openrct2-ui/windows/Map.cpp:1365]: Yes, perhaps. What would be a good name for it? Something like `kCornerIndicatorLength`?

Comment (AaronVanGeffen) [reply src/openrct2-ui/windows/Map.cpp:1365]: On second thought, I believe they should be removed along with the ones previously removed in f7330a2.

**Comment by AaronVanGeffen** [comment]
This is ready for review and testing. I'd like to squash/fixup some commits before merging though.

**Pull request merged**